File: cpp/src/FileMonitorRobot.cpp
#include "../include/FileMonitorRobot.h"
#include "../include/TradingRobot.h"

#include <sys/types.h>
#include <sys/stat.h>
#include <unistd.h>

FileMonitorRobot::FileMonitorRobot(string file, TradingRobot& tradingrobot) : FileMonitor(file), _tradingrobot(tradingrobot)
{
    setLocked(false);
}

FileMonitorRobot::~FileMonitorRobot()
{
}

void FileMonitorRobot::onChange()
{
    cout << "File has changed... Proceeding..." << endl;
    setLocked(true);
    _tradingrobot.onChange();
    setLocked(false);
}

File: ats/src/main/java/fr/automated/trading/systems/tradingrobot/monitor/filemonitoring/FileMonitorRobot.java
package fr.automated.trading.systems.tradingrobot.monitor.filemonitoring;

import fr.automated.trading.systems.tradingrobot.robot.TradingRobot;

import java.io.File;

public class FileMonitorRobot extends FileMonitor {
	
	private final TradingRobot tradingrobot;

	public FileMonitorRobot(File file, TradingRobot tradingrobot) {
		super(file);
		this.tradingrobot = tradingrobot;
	}
	
	@Override
    public void onChange() {
		
		setLocked(true);
		tradingrobot.onChange();
		setLocked(false);
	
	}
}
